FGEGraphics/GraphicsHelpers/Models/ShapeGenerators.cs

```diff
@@ -0,0 +1,237 @@
+            float cosTheta = (float)Math.Cos(theta);
+            for (uint j = 0; j <= sides; j++)
+            {
+                float phi = j * MathHelper.TwoPi / sides;
+                float sinPhi = (float)Math.Sin(phi);
+                float cosPhi = (float)Math.Cos(phi);
+                float centerX = radius * cosPhi;
+                float centerY = 0;
+                float centerZ = radius * sinPhi;
+                float x = (radius + tubeRadius * cosTheta) * cosPhi;
+                float y = tubeRadius * sinTheta;
+                float z = (radius + tubeRadius * cosTheta) * sinPhi;
+                vertices.Add(new Vector3(x, y, z));
+                Vector3 pointOnSurface = new(x, y, z);
+                Vector3 centerToSurface = pointOnSurface - new Vector3(centerX, centerY, centerZ);
+                normals.Add(centerToSurface.Normalized());
+                textureCoords.Add(new Vector2((float)j / sides, (float)i / rings));
+                if (i < rings && j < sides)
+                {
+                    uint currentRow = i * (sides + 1);
+                    uint nextRow = (i + 1) * (sides + 1);
+                    indices[index++] = currentRow + j + 1;
+                    indices[index++] = nextRow + j + 1;
+                    indices[index++] = currentRow + j;
+                    indices[index++] = nextRow + j + 1;
+                    indices[index++] = nextRow + j;
+                    indices[index++] = currentRow + j;
+                }
+            }
+        }
+        return GetModelAfterGenerating(modelEngine, "torus", vertices, normals, textureCoords, indices);
+    }
+
+    /// <summary>Returns a shape model after generating it.</summary>
+    public static Model GetModelAfterGenerating(ModelEngine engine, string name, List<Vector3> vertices, List<Vector3> normals, List<Vector2> texCoords, uint[] indices)
+    {
+        Model3DMesh mesh = new()
+        {
+            Name = name,
+            Vertices = [.. vertices.ConvertAll(v => v.ToLocation().ToNumerics())],
+            Normals = [.. normals.ConvertAll(n => n.ToLocation().ToNumerics())],
+            TexCoords = [.. texCoords.ConvertAll(t => new System.Numerics.Vector2(t.X, t.Y))],
+            Indices = indices,
+            Bones = []
+        };
+        Model3D model3D = new()
+        {
+            RootNode = new Model3DNode()
+            {
+                Name = name,
+                Children = [],
+            },            
+            Meshes = [mesh]
+        };
+        return engine.FromScene(model3D, name);
+    }
+
+    /// <summary>Internal methods for shape generation.</summary>
+    public class InternalMethods
+    {
+        /// <summary>Generates a circle and provides the necessary information.</summary>
+        public static void GenerateCircle(List<Vector3> vecs, List<Vector3> norm, List<Vector2> tc, float radius, uint slices, bool flip, float zC, Action<uint> idxAction)
```

Comment:
As noted in earlier comment, this should not be an `Action` here

Reply:
The problem is I had to make changes to the circle generator to support the changes to the cylinder but this caused an issue the 2D circle generator started generating a flipped triangle so I had to make changes to how I was generating the 2D circle. The action is there to support different types of generation methods for indices.

Reply:
It should just need a flip bool input to handle that?

Reply:
Sure I'll see if I can do that.